cmake/linker/ld/gcc/linker_flags.cmake

```diff
@@ -8,3 +8,7 @@ if (NOT CONFIG_COVERAGE_GCOV)
 endif()
 
 check_set_linker_property(TARGET linker APPEND PROPERTY gprof -pg)
+
+# GCC 11 by default emits DWARF version 5 which cannot be parsed by
+# pyelftools. Can be removed once pyelftools supports v5.
+add_link_options(-gdwarf-4)
```

Comment:
All "Zephyr" executables should already be using DWARF v4 because of 72c2f182bc4170ccffa13a52ad99eb5d0f74adc2.

Are you adding a target other than `zephyr_interface` in your build system?

Reply:
I looked for all instances of TARGET in our cmake files. We use zephyr_library_named() to create some other named libraries. Here is an example of what that looks like

```
zephyr_library_named(ec_shim)
get_property(APP_LTO_PROPERTY TARGET app PROPERTY INTERPROCEDURAL_OPTIMIZATION)
set_property(TARGET ec_shim PROPERTY INTERPROCEDURAL_OPTIMIZATION ${APP_LTO_PROPERTY})
set_property(TARGET ec_shim APPEND PROPERTY COMPILE_DEFINITIONS __REQUIRE_ZEPHYR_GPIOS__)
```

In any case, I can't find a dwarf-5 symbol in any .obj file, but they still appear in the zephyr-pre0.elf file. And all gcc compile commands include -gdwarf-4. The only commands that don't have them are the link steps, and adding it here fixes the problems and I an run the ram_report and rom_report again.

Reply:
Fair enough. Using `add_link_options`, although not ideal, seems to the simplest workaround for that problem.

No major complaint about it since this is only a temporary workaround until the DWARF v5 format is properly supported by Zephyr and its tooling.